### PR TITLE
convert OR to pupa and change to OData API

### PR DESCRIPTION
**Pull request by giantryansaul**
Related to #1442

### CODE REVIEW

**Comment by giantryansaul** [comment]
Related to #1505

**Comment by giantryansaul** [comment]
Amended my commits above to use the correct e-mail address if anyone is wondering about the duplicates.

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,4 @@ openstates.sqlite3
 dump/
 venv
 tmp.*
+.idea
```
Comment (jmcarp) [inline .gitignore]: This can probably go in your global gitignore instead of checking it in here.

File: billy_metadata/or.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                               '2007 - 2008 Interim']
+)
+
+def session_list():
+    from billy.scrape.utils import url_xpath
+    return [x.strip() for x in
+            url_xpath('https://olis.leg.state.or.us/liz/sessions/',
+                      '//a[contains(@href, "/liz/")]/text()')]
+
+def extract_text(doc, data):
```
Comment (jmcarp) [inline billy_metadata/or.py]: Is this used here?

File: billy_metadata/or.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                               '2015-2016 Interim',
+                               '2013 1st Special Session',
+                               '2012 1st Special Session',
+                               '2013 - 2014 Interim',
+                               '2011 - 2012 Interim',
+                               '2009 - 2010 Interim',
+                               '2007 - 2008 Interim']
+)
+
+def session_list():
```
Comment (jmcarp) [inline billy_metadata/or.py]: Can you move this to a method of the `Oregon` class and rename to `get_session_list`?

File: openstates/or/__init__.py [hunk 4]
```diff
@@ -1,136 +1,129 @@
...
-            'slug': '2008S1',
+
+class Oregon(Jurisdiction):
+    division_id = "ocd-division/country:us/state:or"
+    classification = "government"
+    name = "Oregon"
+    url = "https://olis.leg.state.or.us"
+    scrapers = {
+        # Commented out for testing
+        # 'people': ORPersonScraper,
```
Comment (jmcarp) [inline openstates/or/__init__.py]: Need to uncomment these when ready to merge 😄

**Comment by jmcarp** [comment]
Can you add Oregon to `flake8.sh` and fix any flake8 issues that might turn up?

**Comment by giantryansaul** [comment]
@jmcarp For sure, still working on converting all of the site scrapers over to the Odata API, but almost done now, just need to convert votes. I'll add your changes.

**Comment by giantryansaul** [comment]
@jmcarp I feel like the pupa conversion is in a good place, everything seems to be scraped well. Only thing is that this requires a username/password for the name. I'm passing those in as environment variables in the container. How could we go about putting that in the master build?

**Comment by jamesturk** [comment]
We set those as env vars (privately) when we run, we'll go ahead and register for some keys,

could you add a note on how to get them here: http://docs.openstates.org/en/latest/policies/state-api-keys.html

**Comment by jamesturk** [comment]
going to review this as soon as I have some keys 👍

**Comment by giantryansaul** [comment]
Sounds good, I think I'll need to rebase with some recent changes since this branch is getting a little behind master, but I'll wait on your code review.

**Comment by jamesturk** [comment]
OK I have keys now, having some issues trying to run it:

```
Traceback (most recent call last):                                                                                   File "/Users/james/.virtualenvs/pupa/bin/pupa", line 9, in <module>                                                  load_entry_point('pupa==0.6.0', 'console_scripts', 'pupa')()                                                     File "/Users/james/code/openstates/pupa/pupa/cli/__main__.py", line 67, in main                                      subcommands[args.subcommand].handle(args, other)                                                                 File "/Users/james/code/openstates/pupa/pupa/cli/commands/update.py", line 274, in handle                            return self.do_handle(args, other, juris)                                                                        File "/Users/james/code/openstates/pupa/pupa/cli/commands/update.py", line 315, in do_handle                         self.check_session_list(juris)                                                                                   File "/Users/james/code/openstates/pupa/pupa/cli/commands/update.py", line 242, in check_session_list                scraped_sessions = juris.get_session_list()                                                                      File "/Users/james/code/openstates/openstates/openstates/or/__init__.py", line 134, in get_session_list              sessions = OregonLegislatorODataClient(self).all_sessions()                                                      File "/Users/james/code/openstates/openstates/openstates/or/apiclient.py", line 39, in all_sessions                  return self.get('sessions')                                                                                      File "/Users/james/code/openstates/openstates/openstates/or/apiclient.py", line 65, in get                           response = self.scraper.get(url, *requests_args,                                                               AttributeError: 'Oregon' object has no attribute 'get'
```

**Comment by jamesturk** [comment]
Made a couple of small fixes here: https://github.com/openstates/openstates/tree/pupa-or

Will keep looking at it in more detail.

**Comment by jamesturk** [comment]
looking pretty good w/ some fixes to normalize the session #s against the old data and a few other minor things

left to do:

- [ ] bill documents
- [ ] votes: it looks like some votes aren't being picked up, needs further investigation
- [ ] votes: counts aren't being set properly (example: ('HB2277', 'Third reading. Carried by Smith G. Passed.', 0) differ on no_count 2 != 56)

**Comment by giantryansaul** [comment]
Awesome! Are the issues you raised still outstanding? I can probably look into these soon.

**Comment by jamesturk** [comment]
I believe the issues I noted got fixed, if you want to spot check the data
that'd be great, let's just verify it has run (we had a couple of issues w/
our ECS task runner over the past couple of days)

**Pull request merged**